runtime/vm/createramclass.cpp

```diff
@@ -1139,116 +1141,137 @@ fillJITVTableSlot(J9VMThread *vmStruct, UDATA *currentSlot, J9Method *currentMet
-								PORT_ACCESS_FROM_VMC(vmThread);
-								J9UTF8 *classNameUTF = J9ROMCLASS_CLASSNAME(romClass);
-								J9Method *method = (J9Method *)storeValue;
-								J9Class *methodClass = ((J9ConstantPool *)((UDATA)method->constantPool & ~J9_STARTPC_STATUS))->ramClass;
-								J9UTF8 *methodClassNameUTF = J9ROMCLASS_CLASSNAME(methodClass->romClass);
-								J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
-								J9UTF8 *nameUTF = J9ROMMETHOD_NAME(romMethod);
-								J9UTF8 *sigUTF = J9ROMMETHOD_SIGNATURE(romMethod);
-								j9tty_printf(PORTLIB, "\nnot overriding %.*s.%.*s%.*s in class %.*s - already overridden",
-									J9UTF8_LENGTH(methodClassNameUTF), J9UTF8_DATA(methodClassNameUTF),
-									J9UTF8_LENGTH(nameUTF), J9UTF8_DATA(nameUTF), J9UTF8_LENGTH(sigUTF), J9UTF8_DATA(sigUTF),
-									J9UTF8_LENGTH(classNameUTF), J9UTF8_DATA(classNameUTF));
-							}
-#endif
+								J9UTF8_DATA(classNameUTF), J9UTF8_LENGTH(nameUTF), J9UTF8_DATA(nameUTF), J9UTF8_LENGTH(sigUTF), J9UTF8_DATA(sigUTF));
+						}
+					} else {
+						if (ROM_METHOD_ID_TAG != ((UDATA)storeValue & VTABLE_SLOT_TAG_MASK)) {
+							PORT_ACCESS_FROM_VMC(vmThread);
+							J9UTF8 *classNameUTF = J9ROMCLASS_CLASSNAME(romClass);
+							J9Method *method = (J9Method *)storeValue;
+							J9Class *methodClass = ((J9ConstantPool *)((UDATA)method->constantPool & ~J9_STARTPC_STATUS))->ramClass;
+							J9UTF8 *methodClassNameUTF = J9ROMCLASS_CLASSNAME(methodClass->romClass);
+							J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
+							J9UTF8 *nameUTF = J9ROMMETHOD_NAME(romMethod);
+							J9UTF8 *sigUTF = J9ROMMETHOD_SIGNATURE(romMethod);
+							j9tty_printf(PORTLIB, "\nnot overriding %.*s.%.*s%.*s in class %.*s - already overridden",
+											J9UTF8_LENGTH(methodClassNameUTF), J9UTF8_DATA(methodClassNameUTF),
+											J9UTF8_LENGTH(nameUTF), J9UTF8_DATA(nameUTF), J9UTF8_LENGTH(sigUTF), J9UTF8_DATA(sigUTF),
+											J9UTF8_LENGTH(classNameUTF), J9UTF8_DATA(classNameUTF));
 						}
+#endif
 					}
 				}
-			/* Keep checking the rest of the methods in the superclass. */
-			superclassVTableIndex--;
+				/* Keep checking the rest of the methods in the superclass. */
+				superclassVTableIndex--;
+			}
 		}
-	}
 
-	/* If the method did not override any parent method, allocate a new slot for it. */
-	if (!anyOverrides && !isPrivate) {
-		/* allocate vTable slot */
-		vTableWriteIndex = growNewVTableSlot(vTableAddress, vTableWriteIndex, storeValue);
-#if defined(J9VM_TRACE_VTABLE_ACCESS)
-		{
-			PORT_ACCESS_FROM_VMC(vmThread);
-			J9UTF8 *classNameUTF = J9ROMCLASS_CLASSNAME(romClass);
-			j9tty_printf(PORTLIB, "\n<vtbl_init: adding @ index=%d %.*s.%.*s%.*s (0x%x)>", vTableWriteIndex, J9UTF8_LENGTH(classNameUTF),
-					J9UTF8_DATA(classNameUTF), J9UTF8_LENGTH(nameUTF), J9UTF8_DATA(nameUTF), J9UTF8_LENGTH(sigUTF), J9UTF8_DATA(sigUTF), romMethod);
+		/* default method always requires a new slot */
```

Comment:
Can we call this `package` rather than `default` to avoid the confusion with Java 8's default methods aka method implementations in interfaces?

Reply:
Sure.